validate/__main__.py

```diff
@@ -24,39 +24,43 @@
 def main():
     '''Main function and CLI tool'''
     args = parse_args(sys.argv[1:])
-    path = Path(args.path)
     input_type = args.type
     file_type = input_type
-
-    try:
-        path_exists(path)
-    except IOError as err:
-        sys.exit(f"Error: {args.path} {str(err)}")
-
-    try:
-        if input_type in FILE_TYPES_DICT:
-            validate_file(path, input_type)
-        elif input_type == GENERIC_FILE_TYPE:
-            file_type, _ = detect_file_type_and_ext(path)
-            validate_file(path, file_type)
-        elif input_type in DIR_TYPES:
-            validate_dir(path, input_type)
-        elif input_type in CHECKSUM_GEN_TYPES:
-            create_checksum_file(path, input_type)
-    except TypeError as err:
-        sys.exit(f"Error: {args.path} {str(err)}") # raise errors, has implicit exit code
-    except ValueError as err:
-        sys.exit(f"Error: {args.path} {str(err)}")
-    except (IOError, OSError) as err:
-        sys.exit(f"Error: {args.path} {str(err)}")
-
-    print(f"Input: {args.path} is valid {file_type}")
+    err_flag = False
```

Comment:
@aholmes Not sure if this is the best way to do it? I set it to true when there's at least one exception caught throughout the run of the program, so it continues validating the rest of the files and exits at the end with error code 1.

Reply:
Yep this is a good way to do it. Another way is to append errors to a list, and check if the list is empty at the end. The way you did it here allows messages to be printed as it goes (assuming they are flushing).

The only suggestion I have is to name the variable `errored` or something similar. `err_flag` doesn't convey the same meaning that "true" means "an error occurred." This name would be great for an enum of errors though.